api/v1alpha1/sparkcluster_types.go

```diff
@@ -93,3 +93,11 @@ type SparkClusterList struct {
 func init() {
 	SchemeBuilder.Register(&SparkCluster{}, &SparkClusterList{})
 }
+
+// IsIncompatibleVersion checks if the provided instance of SparkCluster struct
+// has a version compatible with the current one.
+func (sc *SparkCluster) IsIncompatibleVersion() bool {
+	// Unfortunately we can't rely on e.g. Spec.TypeMeta.APIVersion, because of
+	// past breaking changes for which the meta information hasn't been updated.
+	return sc.Spec.Driver.Selector == nil
```

Comment:
How did you select this field? I think it can be empty in a valid configuration.